### ISSUE TITLE
Possible use cases

### ISSUE THREAD

**Issue by nhenriksen** [nhenriksen/speakeasy#2]
Tentative strategy for dealing with possible use case scenarios.

Preamble: I think OpenFF parameterization should always occur on whole molecules, not residue fragments.

1) Input: 1 molecule made up of 1 residue.
  - Parameterize with OpenFF and createSystem
  - Create Parmed parm and unique_type
  - save mol2, frcmod, pdb, and leaprc (leaprc deals with getting elements correct in tleap)

2) Input: 1 molecule with 2 or more different residues.
  - Parameterize with OpenFF and createSystem
  - Load system into Parmed and unique_type on the full system parm
  - Identify bonds between each residue, store for use in leaprc
  - Create a separate parm for each residue
  - save mol2, individual frcmod, batch frcmod, full pdb, and leaprc (pdb should take care of bonding)

3) Input: 1 molecule with at least 2 repeating residues
  - Parameterize with OpenFF and createSystem
  - Load system in Parmed and identify repeating residues (by name? check atom names too?)
  - unique_type the full system, but post-process the duplicate residues to have matching types
  - (above will generate unused atom types ... probably should create a work around)
  - create a separate parm for each unique residue
  - save mol2, individual frcmod, batch frcmod, full pdb, and leaprc (pdb should take care of bonding)

4) Input: a batch of molecules, some single, maybe some multi residue
  - Do some combo of above ... but all the parms need to go to unique_type together as batch.

**Comment by nhenriksen** [comment]
Info for console scripts if we want this to also be a shell command line tool:
http://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

**Comment by slochower** [comment]
Yup, I've seen this done before. I don't think it will be too complicated. See here in `setup.py`: https://github.com/greenelab/manubot/blob/27456594c407f3fed35804db4bcf3b5bba88a716/setup.py#L74-L79 which calls a `main()` function: https://github.com/greenelab/manubot/blob/27456594c407f3fed35804db4bcf3b5bba88a716/manubot/manubot.py#L388
